File: JAVA/Practica1.java
import java.lang.Math;

class Punto {
    private double x;
    private double y;

    public Punto(double x, double y) {
        this.x = x;
        this.y = y;
    }

    public double[] coordCartesianas() {
        return new double[]{x, y};
    }

    public double[] coordPolares() {
        double radio = Math.sqrt(x * x + y * y);
        double angulo = Math.toDegrees(Math.atan2(y, x));
        return new double[]{radio, angulo};
    }

    @Override
    public String toString() {
        return String.format("(%.2f, %.2f)", x, y);
    }

    public static void main(String[] args) {
        Punto p1 = new Punto(0, 3);
        System.out.println(p1);
        double[] cartesianas = p1.coordCartesianas();
        System.out.printf("x: %.2f, y: %.2f%n", cartesianas[0], cartesianas[1]);
        double[] polares = p1.coordPolares();
        System.out.printf("r: %.2f, a: %.2f%n", polares[0], polares[1]);
    }
}

File: PYTHON/Practica1.py
import math
#Importamos la libreria math para poder hacer operaciones matematicas y poder agregar graficos
#Esta clase es la clase punto
class Punto:
    def __init__(self, x, y):
        self.x = x
        self.y = y

    def coord_cartesianas(self):
        return self.x, self.y

    def coord_polares(self):
        radio = math.sqrt(self.x * self.x + self.y * self.y)
        angulo = math.atan(self.y / self.x)
        angulo = math.degrees(angulo)
        return radio, angulo

    def __str__(self):
        return "({:.2f},{:.2f})".format(self.x, self.y)

# Ejemplo de uso:
p1 = Punto(0, 3)
print(p1)
x, y = p1.coord_cartesianas()
print(f"x: {x}, y: {y}")
r, a = p1.coord_polares()
print(f"r: {r}, a: {a}")
